equinix/data_source_network_device.go

```diff
@@ -0,0 +1,759 @@
+			Description: neDeviceUserKeyDescriptions["KeyName"],
+		},
+	}
+}
+
+func createDataSourceClusterNodeDetailSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		neDeviceClusterNodeSchemaNames["LicenseFileId"]: {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Sensitive:   true,
+			Description: neDeviceClusterNodeDescriptions["LicenseFileId"],
+		},
+		neDeviceClusterNodeSchemaNames["LicenseToken"]: {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: neDeviceClusterNodeDescriptions["LicenseToken"],
+		},
+		neDeviceClusterNodeSchemaNames["VendorConfiguration"]: {
+			Type:     schema.TypeList,
+			Computed: true,
+			Elem: &schema.Resource{
+				Schema: createDataSourceVendorConfigurationSchema(),
+			},
+			Description: neDeviceClusterNodeDescriptions["VendorConfiguration"],
+		},
+		neDeviceClusterNodeSchemaNames["UUID"]: {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: neDeviceClusterNodeDescriptions["UUID"],
+		},
+		neDeviceClusterNodeSchemaNames["Name"]: {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: neDeviceClusterNodeDescriptions["Name"],
+		},
+	}
+}
+
+func createDataSourceVendorConfigurationSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		neDeviceVendorConfigSchemaNames["Hostname"]: {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: neDeviceVendorConfigDescriptions["Hostname"],
+		},
+		neDeviceVendorConfigSchemaNames["AdminPassword"]: {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Sensitive:   true,
+			Description: neDeviceVendorConfigDescriptions["AdminPassword"],
+		},
+		neDeviceVendorConfigSchemaNames["Controller1"]: {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: neDeviceVendorConfigDescriptions["Controller1"],
+		},
+		neDeviceVendorConfigSchemaNames["ActivationKey"]: {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Sensitive:   true,
+			Description: neDeviceVendorConfigDescriptions["ActivationKey"],
+		},
+		neDeviceVendorConfigSchemaNames["ControllerFqdn"]: {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: neDeviceVendorConfigDescriptions["ControllerFqdn"],
+		},
+		neDeviceVendorConfigSchemaNames["RootPassword"]: {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Sensitive:   true,
+			Description: neDeviceVendorConfigDescriptions["RootPassword"],
+		},
+	}
+}
+func dataSourceNetworkDevice() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceNetworkDeviceRead,
+		Description: "Use this data source to get details of Equinix Network Edge network device with a given Name or UUID",
+		Schema:      createDataSourceNetworkDeviceSchema(),
+	}
+}
+
+func getDeviceByName(deviceName string, conf *Config) (*ne.Device, error) {
+	var devices []ne.Device
+	err := error(nil)
+	devices, err = conf.ne.GetDevices(neDeviceStates)
```

Comment:
We provide searchText="deviceName or device uuid" as query parameters in this GET devices API to search by name or uuid. 
It looks like we forget to put it in the documentation. Will update our public documentation.

Reply:
> Piggyback to what Marques suggest, we also provide status as query parameters to fetch the devices. We use comma to delimit multiple statuses. 

Sure.  to be clear I'm not implementing the API calls directly from the ground up, instead simply using the existing functions 
 from the `ne-go` package:

```go
type Client interface {
       ...
	GetDevice(uuid string) (*Device, error)
	GetDevices(statuses []string) ([]Device, error)
```

So my lookups here will just inherit the correctness of this package and the underlying API, and any improvements done there.